common/src/main/java/io/netty/util/internal/Murmur3.java

```diff
@@ -0,0 +1,464 @@
+                        (toLowerCase((byte) data[intI + 2]) << 16) |
+                        (toLowerCase((byte) data[intI + 1]) << 8) |
+                        toLowerCase((byte) data[intI]);
+            }
+            @Override
+            public int getIntAsBytes(char[] data, long i) {
+                final int intI = (int) i;
+                return toLowerCase((byte) data[intI + 3]) << 24 |
+                        toLowerCase((byte) data[intI + 2]) << 16 |
+                        toLowerCase((byte) data[intI + 1]) << 8 |
+                        toLowerCase((byte) data[intI]);
+            }
+            @Override
+            public long byteArrayBaseOffset() {
+                return 0;
+            }
+        }, DEFAULT_SEED);
+    }
+
+    private Murmur3(NativeTypeAccessor accessor, int seed) {
+        this.accessor = checkNotNull(accessor, "accessor");
+        this.seed = seed;
+    }
+
+    @Override
+    public int hashCodeAsBytes(char[] data, int offset, int len) {
+        int h1 = seed;
+        //----------
+        // body
+        final int numIntsOffset = len & 0xFFFFFFFC;
+        final int end = offset + numIntsOffset;
+        for (int i = offset; i < end; i += 4) {
+            int k1 = accessor.getIntAsBytes(data, i);
+            k1 *= MurmurHash3_C1;
+            k1 = Integer.rotateLeft(k1, MurmurHash3_ROLT1);
+            k1 *= MurmurHash3_C2;
+
+            h1 ^= k1;
+            h1 = Integer.rotateLeft(h1, MurmurHash3_ROLT2);
+            h1 = (h1 * 5) + MurmurHash3_C3;
+        }
+
+        //----------
+        // tail
+        offset += numIntsOffset;
+        int k1 = 0;
+        switch (len & 3) {
+            case 3: k1 = accessor.getByte(data, offset + 2) << 16;
+            case 2: k1 ^= accessor.getByte(data, offset + 1) << 8;
+            case 1: k1 ^= accessor.getByte(data, offset);
+            h1 ^= murmurHash3Tail(k1);
+            default:
+                break;
+        }
+
+        //----------
+        // finalization
+        h1 ^= len;
+        return murmurHash3Fmix(h1);
+    }
+
+    @Override
+    public int hashCodeAsBytes(CharSequence data, int offset, int len) {
+        // CharSequence is slow to access, and so we should try to pick out commonly used known types to get direct
+        // access to a char[] which may be eligible for UNSAFE type operations.
+        if (STRING_VALUE_FIELD_OFFSET != -1) {
+            if (data.getClass().equals(String.class)) {
+                return hashCodeAsBytes(
+                        (char[]) PlatformDependent0.getObject(data, STRING_VALUE_FIELD_OFFSET), offset, len);
+            }
+        }
+        if (STRING_BUILDER_VALUE_FIELD_OFFSET != -1) {
+            if (data.getClass().equals(StringBuilder.class)) {
+                return hashCodeAsBytes(
+                        (char[]) PlatformDependent0.getObject(data, STRING_BUILDER_VALUE_FIELD_OFFSET), offset, len);
+            }
+        }
+        int h1 = seed;
```

Comment:
We don't currently have a use case for this case, correct?  ... just here for completeness?